src/iosrv/sched.c

```diff
@@ -268,16 +270,42 @@ static void
 prune_purge_list(struct dss_xstream *dx)
 {
 	struct sched_info	*info = &dx->dx_sched_info;
+	struct sched_pool_info	*spi;
 	struct purge_item	*pi, *tmp;
+	d_list_t		*rlink;
 	bool			 deleted;
 
 	d_list_for_each_entry_safe(pi, tmp, &info->si_purge_list, pi_link) {
-		deleted = d_hash_rec_delete(info->si_pool_hash, pi->pi_pool_id,
-					    sizeof(uuid_t));
-		if (!deleted)
-			D_ERROR("XS(%d): Purge "DF_UUID" failed.\n",
-				dx->dx_xs_id, DP_UUID(pi->pi_pool_id));
+		rlink = d_hash_rec_find(info->si_pool_hash, pi->pi_pool_id,
+					sizeof(uuid_t));
+		if (rlink == NULL)
+			goto next;
+
+		spi = sched_rlink2spi(rlink);
+		D_ASSERT(spi->spi_ref > 1);
+		d_hash_rec_decref(info->si_pool_hash, rlink);
+		if (spi->spi_req_cnt == 0) {
```

Comment:
only when (spi->spi_req_cnt == 0) will add it to the purge list, so still possible add req after that?

Reply:
Yes, that's possible. Current implementation is that: on schedule cycle start, it prune the purge list first to delete the 'spi' added by previous cycle, then it iterates the spi hash, and some spi could be added into purge list for next cycle. So a 'spi' added in previous cycle could be referenced by new request in next cycle.